tests/Unit/NamespaceScopeVariables/global.cpp

```diff
@@ -0,0 +1,53 @@
+// RUN: %hc %s -o %t.out && %t.out
+
+#include "Inputs/test_parameters.hpp"
+
+#include <hc.hpp>
+
+#include <algorithm>
+#include <cstdlib>
+
+int global_scalar;
+int global_array[array_size];
+
+int main()
+{
+    using namespace hc;
+    using namespace std;
+
+    global_scalar = rand();
+    generate_n(global_array, array_size, rand);
+
+    array_view<int> read_scalar{1};
+    array_view<int> read_array(array_size);
+
+    parallel_for_each(hc::extent<1>{1}, [=](index<1>) [[hc]] {
+        read_scalar[0] = global_scalar;
+        for (auto i = 0u; i != array_size; ++i) read_array[i] = global_array[i];
+    });
+
+    if (read_scalar[0] != global_scalar) return EXIT_FAILURE;
+    if (!equal(global_array, global_array + array_size, read_array.data())) {
+        return EXIT_FAILURE;
+    }
+
+    #if false // 10/09/2017 - writes to globals are not correctly observed.
```

Comment:
any idea why this doesn't work

Reply:
@scchan not at this moment, and unless there is a strong need for that now (I do not believe that is the case, but I may be wrong) I would punt on it. Mainly due to time constraints and it working in HC2. It is your call though.